### ISSUE TITLE
After a change is merged, travis build for merge fails due to listing as "detached HEAD"

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8417]
|[<img alt="chellygel's avatar" src="https://avatars.githubusercontent.com/u/634036?s=50" width="50" height="50">](https://github.com/chellygel)| @chellygel reported|
|-|-|
|Trac ID|trac#8417|
|Type|defect|
|Created|2016-06-04 01:12:24Z|
|Branch|https://github.com/twisted/twisted/tree/fix-topfile-check-for-trunk-8417|

Once a change has been approved and merged, TravisCI build for merge commit of trunk will fail if it is listed as "detached HEAD" and will say the failure, "No topfile found. Have you commited it?"

Instead, it should skip the topfile check.

<details><summary>Searchable metadata</summary>

```
trac-id__8417 8417
type__defect defect
reporter__chellygel chellygel
priority__normal normal
milestone__None None
branch__fix_topfile_check_for_trunk_8417 fix-topfile-check-for-trunk-8417
branch_author__ 
status__closed closed
resolution__fixed fixed
component__release_management release management
keywords__None None
time__1465002744472530 1465002744472530
changetime__1465031075921392 1465031075921392
version__None None
owner__cyli cyli
cc__radix
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)|@cyli removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph set owner to @cyli|
|-|-|

Thank you for fixing this test failure that makes our build say "failing", cyli.

  1. There's a minor coding-standard-compliance issue; `test_diff_from_from_trunk_no_topfiles` and `test_no_change_from_trunk` should be `test_diffFromFromTrunkNoTopfiles` and `test_noChangeFromTrunk`, respectively.
  2. Also I think there are too many `from`s in the first one.

Please address these minor issues and land.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_6"></a>|@cyli set status to `closed`|
|-|-|

In changeset bf436c724c94268ef470699577356ea887c503cd
```
#!CommitTicketReference repository="" revision="bf436c724c94268ef470699577356ea887c503cd"
Merge pull request #137 from twisted/fix-topfile-check-for-trunk-8417: Do not check for topfiles if current git HEAD has no diffs from trunk

Author: cyli
Reviewer: glyph
Fixes: #8417

Do not check for topfiles if current git HEAD has no diffs from trunk (#8417)
```
